autograder-core/src/main/java/de/firemage/autograder/core/integrated/IntegratedAnalysis.java

```diff
@@ -39,16 +52,50 @@ public IntegratedAnalysis(UploadedFile file, Path jar, Path tmpPath, Consumer<Li
     public void runDynamicAnalysis(Path tests, Consumer<LinterStatus> statusConsumer)
         throws RunnerException, InterruptedException {
         try {
-            DockerConsoleRunner runner = new DockerConsoleRunner(
-                Path.of(this.getClass().getResource("/executor.jar").toURI()),
-                Path.of(this.getClass().getResource("/agent.jar").toURI()),
-                tests,
-                this.tmpPath);
+            DockerConsoleRunner runner = new DockerConsoleRunner(toPath(this.getClass().getResource("/executor.jar")), toPath(this.getClass().getResource("/agent.jar")), tests, this.tmpPath);
             List<TestRunResult> results = runner.runTests(this.staticAnalysis, this.jar, statusConsumer);
             this.dynamicAnalysis = new DynamicAnalysis(results);
-        } catch (URISyntaxException e) {
+        } catch (URISyntaxException | IOException e) {
             throw new RunnerException(e);
+        } finally {
+            closeOpenFileSystems();
+        }
+    }
+
+    private Path toPath(URL resource) throws URISyntaxException, IOException {
+        if (resource == null) {
+            throw new IllegalArgumentException("URL is null");
+        }
+        URI uri = resource.toURI();
+        if (!uri.toString().contains("!")) {
+            return Path.of(uri);
+        }
+        // See https://stackoverflow.com/questions/22605666/java-access-files-in-jar-causes-java-nio-file-filesystemnotfoundexception
+        String[] path = uri.toString().split("!", 2);
+        @SuppressWarnings("resource") FileSystem fs = createFileSystem(path[0]);
```

Comment:
https://github.com/Feuermagier/autograder/pull/25/files#diff-624cfed3224cd3668e334c207219d10e516920ff3039e75faf40c72b1ce8715dR90 handles the close operation of `fs`